src/gpuarray_array.c

```diff
@@ -1117,8 +1121,7 @@ int GpuArray_is_c_contiguous(const GpuArray *a) {
   int i;
 
   for (i = a->nd - 1; i >= 0; i--) {
-    if (a->dimensions[i] != 1 && a->strides[i] != size)
-      return 0;
+    if (a->strides[i] != size) return 0;
```

Comment:
If I understand the intent of the code `shape = (1, 3, 1, 1)` and `strides = (4, 4, 12, 12)` would pass as C-contiguous when it is obviously not.  This is what I'm trying to fix.

Otherwise I don't get it since the check is written in a confusing way that completely changes the way it was done before.